Sprint-1/stretch/aoc-2018-day1/solution.js

```diff
@@ -0,0 +1,44 @@
+const fs = require("fs");
+const path = require("path");
+
+// Get the current directory of the script
+const filePath = path.join(__dirname, "./input.txt");
+
+fs.readFile(filePath, "utf8", (err, data) => {
+  if (err) {
+    console.error("Error reading file:", err);
+    return;
+  }
+
+  const changes = data.split("\n").filter((line) => line.trim() !== "");
+
+  // Part One: Calculate the final frequency
+  const finalFrequency = calculateFrequency(changes);
+  console.log("Final frequency after all changes:", finalFrequency);
+
+  // Part Two: Find the first repeated frequency
+  const firstRepeatedFrequency = findFirstRepeatedFrequency(changes);
+  console.log("First repeated frequency:", firstRepeatedFrequency);
+});
+
+function calculateFrequency(changes) {
+  //   let frequency = 0;
+  //   for (const change of changes) {
+  //     frequency += parseInt(change, 10);
+  //   }
+  return changes.reduce((acc, change) => acc + Number(change), 0);
+}
+function findFirstRepeatedFrequency(changes) {
+  let frequency = 0;
+  const seenFrequencies = new Set([frequency]);
+
+  while (true) {
+    for (const change of changes) {
+      frequency += Number(change);
```

Comment:
What if the values in the input file are all `+1`? That would means the frequency value won't repeat.

Reply:
If the values in the input array are all +1,  the frequency will never repeat because the frequency will keep increasing indefinitely.

Reply:
Would the loop stop in such case? How should you avoid an infinite loop?

Reply:
By adding a limit on the number of iterations, we initialize let iterations = 0; and set const maxIterations = 100000;. Then, we use a conditional while loop with while (iterations <  maxIterations), and increment iterations++ after each iteration.

Reply:
I just saw the spec for step 2. So Your code is good.

Reply:
Thanks